cmd/add-resources.go

```diff
@@ -95,17 +96,26 @@ var resourcesAddCmd = &cobra.Command{
 	},
 }
 
-func colonMappingsToPortMappings(mappings []string) *models.AccessResourcePortMapping {
-	publicPorts := make([]string, len(mappings))
-	internalPorts := make([]string, len(mappings))
-	for i, mapping := range mappings {
-		parts := strings.SplitN(mapping, ":", 2)
-		publicPorts[i] = strings.TrimSpace(parts[0])
-		internalPorts[i] = strings.TrimSpace(parts[len(parts)-1])
+func colonMappingToPortMapping(mapping string) *models.AccessResourcePortMapping {
+	var protocol string
+	internalPorts := []string{}
+	parts := strings.SplitN(mapping, ":", 3)
+	publicPorts := strings.Split(strings.TrimSpace(parts[0]), ",")
+
+	if len(parts) > 1 {
+		internalPorts = strings.Split(strings.TrimSpace(parts[1]), ",")
+	}
+
+	if len(parts) > 2 {
+		protocol = strings.TrimSpace(parts[2])
+	} else {
+		protocol = "tcp"
 	}
```

Comment:
You can try to use array destructuring to make the code easier to read.

Reply:
not doable in go